usr/share/rear/lib/layout-functions.sh

```diff
@@ -992,4 +992,207 @@ function apply_layout_mappings() {
+        # partition then get its end
+        LogPrint "Disk '$disk': creating a temporary partition to find out the end of the disk"
+        parted -s -m $disk mkpart "$name" ${startB}B 100%
+        local num
+        read num endB <<< $( parted -s -m $disk unit B print | tail -1 | awk -F ':' '{ print $1, $3 }' | sed 's/\([0-9]*\)B$/\1/' )
+        LogPrint "Disk '$disk': last allocatable byte on disk is '$endB'"
+        LogPrint "Disk '$disk': deleting the temporary partition number $num"
+        parted -s -m $disk rm $num
+    fi
+
+    partitions_to_resize+=( $number $endB )
+
+    local -i logical_sector_size=$( parted -m -s $disk unit B print | awk -F ':' "\$1 == \"$disk\" { print \$4 }" )
+
+    if [[ "$disk_label" != "msdos" ]] || [[ "$name" != "logical" ]] ; then
+        local -i i=$last_partition_number+1
+        while [[ $i -lt $number ]] ; do
+            local partname="dummy$i"
+            [[ "$disk_label" != "msdos" ]] || partname="primary"
+            dummy_partitions_to_delete+=( $i )
+            LogPrint "Disk '$disk': creating partition number $i with name '$partname' (will be deleted later)"
+            parted -s -m $disk mkpart "$partname" "${endB}B" "${endB}B"
+            my_udevsettle
+            let endB-=$logical_sector_size
+            if [[ $endB -lt $startB ]] ; then
+                # Not enough space left for real partition: this happens if the
+                # partition to create is very small and we cannot create dummy
+                # temporary partitions!
+                # e.g. Partition table starts at partition number 2 and partition 2
+                # has only 1 cylinder
+                Error "Disk '$disk': Cannot create partition number $number, the partition is too small to create dummy partitions to work around parted's limitation (parted can only create consecutive partitions)"
+            fi
+            let i++
+        done
+    fi
+
+    LogPrint "Disk '$disk': creating partition number $number with name '$name'"
+    parted -s $disk mkpart "$name" "${startB}B" "${endB}B"
+    my_udevsettle
+
+    last_partition_number=$number
+}
+
+delete_dummy_partitions_and_resize_real_ones() {
+    if [[ ${#dummy_partitions_to_delete[@]} -eq 0 ]] ; then
+        partitions_to_resize=()
+        current_disk=""
+        disk_label=""
+        last_partition_number=0
+        return 0
+    fi
+
+    # Delete dummy partitions
+    local -i num
+    for num in ${dummy_partitions_to_delete[@]} ; do
+        LogPrint "Disk '$current_disk': deleting dummy partition number $num"
+        parted -s -m $current_disk rm $num
+    done
+    dummy_partitions_to_delete=()
+    my_udevsettle
+
+    # Resize previously shrinked partitions (to make place for dummy
+    # partitions) to expected size
+    local -i endB
+    while read num endB ; do
+        LogPrint "Disk '$current_disk': resizing partition number $num to original size"
+        parted -s -m $current_disk resizepart $num "${endB}B"
```

Comment:
`parted ... resizepart ...` does not work on older systems.

On my SLES11-SP4 system:
```
# parted --help
...
  resize NUMBER START END                  resize partition NUMBER and its file
        system
```

On my SLES12-SP4 system:
```
# parted --help
..
  resize NUMBER END                        change end sector of partition NUMBER
```

On my SLES15-like openSUSE Leap 15.0 system:
```
# parted --help
...
  resizepart NUMBER END                    resize partition NUMBER
```